services/motion/builtin/moveRequest.go

```diff
@@ -0,0 +1,212 @@
+package builtin
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"time"
+
+	geo "github.com/kellydunn/golang-geo"
+	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/components/base"
+	"go.viam.com/rdk/components/base/kinematicbase"
+	"go.viam.com/rdk/motionplan"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/services/motion"
+	"go.viam.com/rdk/spatialmath"
+)
+
+// moveRequest is a structure that contains all the information necessary for to make a move call.
+type moveRequest struct {
+	planRequest        *motionplan.PlanRequest
+	actuator           kinematicbase.KinematicBase
+	execute            func(context.Context, motionplan.Plan) moveResponse
+	position, obstacle *replanner
+}
+
+// plan creates a plan using the currentInputs of the robot and the moveRequest's planRequest.
+func (mr *moveRequest) plan(ctx context.Context) (motionplan.Plan, error) {
+	inputs, err := mr.actuator.CurrentInputs(ctx)
+	if err != nil {
+		return make(motionplan.Plan, 0), err
+	}
+	// TODO: this is really hacky and we should figure out a better place to store this information
+	if len(mr.actuator.Kinematics().DoF()) == 2 {
+		inputs = inputs[:2]
+	}
+	mr.planRequest.StartConfiguration = map[string][]referenceframe.Input{mr.actuator.Kinematics().Name(): inputs}
+	return motionplan.PlanMotion(ctx, mr.planRequest)
+}
+
+// newMoveOnGlobeRequest instantiates a moveRequest intended to be used in the context of a MoveOnGlobe call.
+func (ms *builtIn) newMoveOnGlobeRequest(
+	ctx context.Context,
+	componentName resource.Name,
+	destination *geo.Point,
+	movementSensorName resource.Name,
+	obstacles []*spatialmath.GeoObstacle,
+	motionCfg *motion.MotionConfiguration,
+	extra map[string]interface{},
+) (*moveRequest, error) {
+	kinematicsOptions := kinematicbase.NewKinematicBaseOptions()
+	if motionCfg.LinearMPerSec != 0 {
+		kinematicsOptions.LinearVelocityMMPerSec = motionCfg.LinearMPerSec * 1000
+	}
+	if motionCfg.AngularDegsPerSec != 0 {
+		kinematicsOptions.AngularVelocityDegsPerSec = motionCfg.AngularDegsPerSec
+	}
+	if motionCfg.PlanDeviationMM != 0 {
+		kinematicsOptions.PlanDeviationThresholdMM = motionCfg.PlanDeviationMM
+	}
+	kinematicsOptions.GoalRadiusMM = motionCfg.PlanDeviationMM
+	kinematicsOptions.HeadingThresholdDegrees = 8
+
+	// build the localizer from the movement sensor
+	movementSensor, ok := ms.movementSensors[movementSensorName]
+	if !ok {
+		return nil, resource.DependencyNotFoundError(movementSensorName)
+	}
+	origin, _, err := movementSensor.Position(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// add an offset between the movement sensor and the base if it is applicable
+	baseOrigin := referenceframe.NewPoseInFrame(componentName.ShortName(), spatialmath.NewZeroPose())
+	movementSensorToBase, err := ms.fsService.TransformPose(ctx, baseOrigin, movementSensor.Name().ShortName(), nil)
+	if err != nil {
+		movementSensorToBase = baseOrigin
+	}
```

Comment:
Could we add a comment that this assumes that the movement sensor is at the center of the base if the movement sensor is not able to be looked up by the frame system?

Reply:
Added